BRANCHES.md

```diff
@@ -28,11 +28,28 @@ undocumented features, then you should be able to re-compile it without
 modification with any later release x.y'.z' with the same major version
 number, and your code will still build, be secure, and work.
 
-There are rare exceptions: code that was relying on something that became
-insecure in the meantime (for example, crypto that was found to be weak) may
-need to be changed. In case security comes in conflict with backwards
-compatibility, we will put security first, but always attempt to provide a
-compatibility option.
+Note that new releases of Mbed TLS may extend the API. Here are some
```

Comment:
Pre-existing but line 19 shouldn't it be: "we maintain
API compatibility in the `master` branch between MINOR version changes." ?

Reply:
Looks correct to me: between two changes of the major version (e.g. between the change from 1.3 to 2.0 and the change from 2.28 to 3.0), the API remains backward-compatible.

We maintain API compatibility *on* minor version changes, not just *between* minor version changes.

Reply:
I feel like "between" here can be interpreted in two ways:

1. "During the interval of time starting at 2.0.0 and ending before 3.0.0, we maintain..." -> _between_ those two points in time, we maintain... -> between major version changes, we maintain...
2. "Version 2.27 is backward compatible with 2.26" -> there's backwards compat _between_ those two versions -> between minor versions.

I think the fact that we're saying "between version changes" rather than "between versions" hints towards the first interpretation.

In general, I think it's important to reduce ambiguity, but in that case there's already a link to a detailed explanation, and we spell things in full in the first paragraph of the next section, so I'm not sure it's worth spending too much time on this.

Reply:
ok got it.

Reply:
I think this is misleading: I interpret saying "we maintain compatibility between major versions" as meaning, e.g., 2.x is API-compatible with 3.x. Maybe the following is clearer?

... we maintain API compatibility in the `master` branch across minor version changes, but not when the major version changes.

I'll approve as-is but would be in favour of addressing this.

Reply:
I've reworded the sentence to avoid the word “between” and added a formulation in terms of version numbers.